ansible/configs/ansible-middleware-workshop/files/cloud_providers/azure_cloud_template.j2

```diff
@@ -0,0 +1,428 @@
+            "name": "[concat(variables('subzone'), '/', '{{instancename}}')]",
+            "apiVersion": "2016-04-01",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/publicIPAddresses/', '{{instancename}}-PublicIP')]",
+                "[resourceId('Microsoft.Network/dnsZones/', variables('subzone'))]",
+            ],
+            "properties": {
+                "TTL": 3600,
+                "ARecords": [
+                    {
+                        "ipv4Address": "[reference('{{instancename}}-PublicIP').ipAddress]"
+                    }
+                ]
+            }
+        },
+        {
+            "apiVersion" : "2017-04-01",
+            "type" : "Microsoft.Network/publicIPAddresses",
+            "name" : "{{instancename}}-PublicIP",
+            "location" : "[resourceGroup().location]",
+            "properties" : {
+                "publicIPAllocationMethod" : "Static",
+                "dnsSettings" : {
+                    "domainNameLabel" : "{{instancename}}-{{guid}}"
+                }
+            }
+        },
+        {
+            "apiVersion" : "2017-04-01",
+            "type" : "Microsoft.Network/networkInterfaces",
+            "name" : "{{instancename}}-Interface",
+            "location" : "[resourceGroup().location]",
+            "dependsOn" : [
+                "[resourceId('Microsoft.Network/publicIPAddresses/', '{{instancename}}-PublicIP')]",
+                "[resourceId('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]"
+            ],
+            "properties" : {
+                "ipConfigurations" : [
+                    {
+                        "name" : "ipconfig1",
+                        "properties" : {
+                            "privateIPAllocationMethod" : "Dynamic",
+                            "publicIPAddress" : {
+                                "id" : "[resourceId('Microsoft.Network/publicIPAddresses','{{instancename}}-PublicIP')]"
+                            },
+                            "subnet" : {
+                                "id" : "[variables('subnetRef')]"
+                            }
+                        }
+                    }
+                ]
+            }
+        },
+{% endif %}
+        {
+            "apiVersion" : "2017-03-30",
+            "type" : "Microsoft.Compute/virtualMachines",
+            "name" : "{{instancename}}",
+            "location" : "[resourceGroup().location]",
+            "dependsOn" : [
+                "[resourceId('Microsoft.Network/networkInterfaces/', '{{instancename}}-Interface')]"
+            ],
+            "tags": {
+                "Name": "{{instancename}}",
+                "internaldns": "{{instancename}}.{{chomped_zone_internal_dns}}",
+                "owner": "{{ email | default('unknownuser') }}",
+                "Project": "{{project_tag}}",
+{% for tag in instance['tags'] %}
+                "{{tag['key']}}": "{{tag['value']}}",
+{% endfor %}
```

Comment:
Likewise for indentation:
```
{%   for tag in instance['tags'] %}
                "{{tag['key']}}": "{{tag['value']}}",
{%   endfor %}
```

Reply:
Thanks @jkupferer, this file was copied from the configs/three-tier-app config, would you like me to make these changes there also?